ocaml/xapi/sm.ml

```diff
@@ -98,9 +91,7 @@ let sr_detach dconf driver sr =
     (fun ()->
        debug "sr_detach" driver (sprintf "sr=%s" (Ref.string_of sr));
        let call = Sm_exec.make_call ~sr_ref:sr dconf "sr_detach" [] in
-       Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call));
-  Threadext.Mutex.execute sr_content_type_cache_m
-    (fun () -> Hashtbl.remove sr_content_type_cache sr)
+       Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call))
```

Comment:
Yes but you also removed the cache, so it's quite a safe change. I'd rather see if anything else (external) was using exposed functions that you removed from the interface.